Software/lib/GNSE_BSP/GNSE_bsp_gpio.c

```diff
@@ -314,17 +314,24 @@ int32_t GNSE_BSP_LS_DeInit(Load_Switch_TypeDef loadSwitch)
 }
 
 /**
-  * @brief  Turns selected Load Switch On.
+  * @brief  Turns selected Load Switch On with settling time delay.
   * @param  loadSwitch: Specifies the Load Switch to be set on.
   *         This parameter can be one of the following values:
   *            @arg LOAD_SWITCH1
   *            @arg LOAD_SWITCH2
   *            @arg LOAD_SWITCH3
+  * @param  delay_ms: Delay in milliseconds for peripheral to turn on.
+  *         This parameter includes these constants as examples:
+  *            LOAD_SWITCH_ON_TIME_MS
+  *            ACCELEROMETER_ON_TIME_MS
+  *            FLASH_ON_TIME_MS
+  *            SECURE_ELEMENT_ON_TIME_MS
   * @return GNSE_BSP status
   */
-int32_t GNSE_BSP_LS_On(Load_Switch_TypeDef loadSwitch)
+int32_t GNSE_BSP_LS_On(Load_Switch_TypeDef loadSwitch, uint8_t delay_ms)
```

Comment:
This changes can create issues on the long run
There is a reason why BSP functions don't have delays in them, and this is because it's a platform/application dependent.

In an OS, you would delay by invoking vTaskDelay(), and while HAL_Delay can be overridden like how i do it [here](https://github.com/TheThingsIndustries/generic-node-se/blob/14385d8f044b19e2d616d80608bc60fe60a9582f/Software/app/freertos_lorawan/sys_app.c#L186) 
 you can see that hiding this delay or building it as part of the BSP can be problematic